src/aggregation/agg_req_with_accessor.rs

```diff
@@ -0,0 +1,126 @@
+//! This will enhance the request tree with access to the fastfield and metadata.
+
+use super::{
+    agg_req::Aggregations, bucket::RangeAggregationReq, Aggregation, BucketAggregationType,
+    MetricAggregation, VecWithNames,
+};
+use crate::{fastfield::DynamicFastFieldReader, SegmentReader, TantivyError};
+use std::collections::HashMap;
+
+//pub type AggregationsWithAccessor = HashMap<String, AggregationWithAccessor>;
+pub type AggregationsWithAccessor = VecWithNames<AggregationWithAccessor>;
+
+/// Aggregation tree with fast field accessors.
+///
+#[derive(Clone)]
+pub enum AggregationWithAccessor {
+    Bucket(BucketAggregationWithAccessor),
+    Metric(MetricAggregationWithAccessor),
+}
+
+impl AggregationWithAccessor {
+    pub fn as_bucket(&self) -> &BucketAggregationWithAccessor {
```

Comment:
nitpick: Can we return `Option<&BucketAggregationWithAccessor>`?

This is mostly to align with what is done in `serde_json::Value::as_*` and in `tantivy::schema::Value::as_*` prototype.